src/python/pants/engine/process.py

```diff
@@ -508,7 +522,7 @@ async def find_binary(request: BinaryPathRequest) -> BinaryPaths:
 
     search_path = create_path_env_var(request.search_path)
     result = await Get(
-        FallibleProcessResult,
+        ProcessResult,
```

Comment:
I don't think this is safe.

```
▶ command -v python4.2; echo $?
1
```

In fact, if you're up for it, a simple test to ensure we can handle a non-existent binary would be very helpful.